File: app/src/main/java/com/veganbeauty/app/features/account/notification/NotificationActionReceiver.java
package com.veganbeauty.app.features.account.notification;

import android.app.NotificationManager;
import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class NotificationActionReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (context == null || intent == null) return;
        String action = intent.getAction();
        int notificationId = intent.getIntExtra("extra_notification_id", -1);
        
        if ("com.veganbeauty.app.ACTION_SKIP".equals(action) && notificationId != -1) {
            NotificationManager notificationManager = (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
            if (notificationManager != null) {
                notificationManager.cancel(notificationId);
            }
        }
    }
}

File: app/src/main/java/com/veganbeauty/app/features/account/notification/NotificationActionReceiver.kt
package com.veganbeauty.app.features.account.notification

import android.app.NotificationManager
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class NotificationActionReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context?, intent: Intent?) {
        if (context == null || intent == null) return
        val action = intent.action
        val notificationId = intent.getIntExtra("extra_notification_id", -1)
        
        if (action == "com.veganbeauty.app.ACTION_SKIP" && notificationId != -1) {
            val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
            notificationManager.cancel(notificationId)
        }
    }
}
